packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.js

```diff
@@ -0,0 +1,131 @@
+import React, { memo } from 'react';
+import PropTypes from 'prop-types';
+import { View, KeyboardAvoidingView } from 'react-native';
+import { SafeAreaView } from 'react-navigation';
+import { Formik } from 'formik';
+import {
+  ModalView,
+  Switch,
+  TextInput,
+  Button,
+  styled,
+  FlexedView,
+  PaddedView,
+  H3,
+  H5,
+  Avatar,
+} from '@apollosproject/ui-kit';
+
+const FlexedSafeAreaView = styled({
+  flex: 1,
+})(SafeAreaView);
+
+const ShrinkingView = styled(({ theme }) => ({
+  flex: 1,
+  paddingTop: theme.sizing.baseUnit,
+}))(KeyboardAvoidingView);
+
+const HeaderView = styled({
+  alignItems: 'center',
+})(View);
+
+const TitleView = styled(({ theme }) => ({
+  marginTop: theme.sizing.baseUnit * 0.5,
+  marginBottom: theme.sizing.baseUnit * 2,
+}))(View);
+
+const StyledTextInput = styled({
+  height: '100%',
+})(TextInput);
+
+const StyledButton = styled({
+  borderRadius: 0,
+})(Button);
+
+const BottomView = styled({
+  justifyContent: 'flex-end',
+})(FlexedView);
+
+const SwitchContainer = styled({
+  width: '70%',
+  alignSelf: 'center',
+  flexDirection: 'row',
+  justifyContent: 'center',
+})(PaddedView);
+
+const SwitchLabel = styled(({ theme }) => ({
+  color: theme.colors.primary,
+}))(H5);
+
+const SwitchLabelContainer = styled(({ theme }) => ({
+  justifyContent: 'center',
+  marginLeft: theme.sizing.baseUnit,
+}))(View);
+
+const AddPrayerForm = memo(({ imgSrc, title, btnLabel, ...props }) => (
+  <Formik
+    initialValues={{ prayer: '', anonymous: false }}
+    onSubmit={(values) => console.log(values)}
+  >
+    {({ handleChange, handleBlur, handleSubmit, values }) => (
+      <ModalView {...props}>
+        <FlexedSafeAreaView forceInset={{ top: 'always', bottom: 'never' }}>
+          <ShrinkingView behavior={'padding'}>
+            <HeaderView>
+              {/* TODO: margin will be fixed pending core #667 */}
+              <Avatar
+                source={values.anonymous ? null : imgSrc}
+                size={'medium'}
+              />
+              <TitleView>
+                <H3>{title}</H3>
+              </TitleView>
+            </HeaderView>
+            {/* TODO: add underline={false} prop to TextInput, 
```

Comment:
You can go ahead and add this prop now, fwiw. It won't do anything yet, but then you won't have to remember to make the change once https://github.com/ApollosProject/apollos-prototype/issues/629 is merged 😄